pkg/controllers/disruption/helpers.go

```diff
@@ -317,7 +317,14 @@ func worstLaunchPrice(ofs []cloudprovider.Offering, reqs scheduling.Requirements
 	// We prefer to launch spot offerings, so we will get the worst price based on the node requirements
 	if reqs.Get(v1beta1.CapacityTypeLabelKey).Has(v1beta1.CapacityTypeSpot) {
 		spotOfferings := lo.Filter(ofs, func(of cloudprovider.Offering, _ int) bool {
```

Comment:
Can we leverage `ofs.Compatible` here? Something along the lines of:
```go
spotOfferings := ofs.Compatible(scheduling.NewRequirements(scheduling.NewRequirement(v1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, v1beta1.CapacityTypeSpot)))
```
Same for the on-demand case.